-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-42756/37392

ERROR: Build errors found during clang-tidy run.
```
/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02808/el8_amd64_gcc12/external/gcc/12.3.1-40d504be6370b5a30e3947a6e575ca28/lib/gcc/x86_64-redhat-linux-gnu/12.3.1/../../../../include/c++/12.3.1/bits/stl_vector.h:423:11: note: template is declared here
--
DQM/HcalTasks/plugins/HcalMLTask.cc:240:32: error: use of class template 'std::vector' requires template arguments [clang-diagnostic-error]
  std::vector<std::vector<std::vector>> digiHcal3DHist_ANOMALY_FLAG_HB =
                               ^
/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02808/el8_amd64_gcc12/external/gcc/12.3.1-40d504be6370b5a30e3947a6e575ca28/lib/gcc/x86_64-redhat-linux-gnu/12.3.1/../../../../include/c++/12.3.1/bits/stl_vector.h:423:11: note: template is declared here
--
DQM/HcalTasks/plugins/HcalMLTask.cc:242:32: error: use of class template 'std::vector' requires template arguments [clang-diagnostic-error]
  std::vector<std::vector<std::vector>> digiHcal3DHist_ANOMALY_FLAG_HE =
                               ^
/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02808/el8_amd64_gcc12/external/gcc/12.3.1-40d504be6370b5a30e3947a6e575ca28/lib/gcc/x86_64-redhat-linux-gnu/12.3.1/../../../../include/c++/12.3.1/bits/stl_vector.h:423:11: note: template is declared here
--
DQM/HcalTasks/plugins/OnlineDQMDigiAD_cmssw.h:94:32: error: use of class template 'std::vector' requires template arguments [clang-diagnostic-error]
  std::vector<std::vector<std::vector>> ONNXOutputToDQMHistMap(const std::vector<std::vector> &ad_model_output_vectors,
                               ^
/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02808/el8_amd64_gcc12/external/gcc/12.3.1-40d504be6370b5a30e3947a6e575ca28/lib/gcc/x86_64-redhat-linux-gnu/12.3.1/../../../../include/c++/12.3.1/bits/stl_vector.h:423:11: note: template is declared here
--
DQM/HcalTasks/plugins/OnlineDQMDigiAD_cmssw.h:94:87: error: use of class template 'std::vector' requires template arguments [clang-diagnostic-error]
  std::vector<std::vector<std::vector>> ONNXOutputToDQMHistMap(const std::vector<std::vector> &ad_model_output_vectors,
                                                                                      ^
/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02808/el8_amd64_gcc12/external/gcc/12.3.1-40d504be6370b5a30e3947a6e575ca28/lib/gcc/x86_64-redhat-linux-gnu/12.3.1/../../../../include/c++/12.3.1/bits/stl_vector.h:423:11: note: template is declared here
--
gmake: *** [config/SCRAM/GMake/Makefile.coderules:129: code-checks] Error 2
gmake: *** [There are compilation/build errors. Please see the detail log above.] Error 2
```

--- a/DQM/HcalTasks/plugins/HcalMLTask.cc
+++ b/DQM/HcalTasks/plugins/HcalMLTask.cc
@@ -238,9 +238,9 @@ void HcalMLTask::globalEndLuminosityBlock(edm::LuminosityBlock const& lb, edm::E
                                                                                            (float)flagDecisionThr);
 
   std::vector<std::vector<std::vector<float>>> digiHcal3DHist_ANOMALY_FLAG_HB =
-      dqmadObj_HB->ONNXOutputToDQMHistMap(ad_HBmodel_output_vectors, 7);
+      dqmadObj_HB->ONNXOutputToDQMHistMap(ad_HBmodel_output_vectors, 4, 64, 7);
   std::vector<std::vector<std::vector<float>>> digiHcal3DHist_ANOMALY_FLAG_HE =
-      dqmadObj_HE->ONNXOutputToDQMHistMap(ad_HEmodel_output_vectors, 7);
+      dqmadObj_HE->ONNXOutputToDQMHistMap(ad_HEmodel_output_vectors, 7, 64, 7);
 
   int NHB_MLbadflags_ = 0, NHE_MLbadflags_ = 0;
   for (const auto& plane : digiHcal3DHist_ANOMALY_FLAG_HB)

--- a/DQM/HcalTasks/plugins/OnlineDQMDigiAD_cmssw.cc
+++ b/DQM/HcalTasks/plugins/OnlineDQMDigiAD_cmssw.cc
@@ -136,12 +136,21 @@ std::vector<float> OnlineDQMDigiAD::Serialize2DVector(const std::vector<std::vec
 
 std::vector<std::vector<float>> OnlineDQMDigiAD::Map1DTo2DVector(const std::vector<float> &input_1d_vec,
                                                                  const int numSplits) {
+  if (numSplits <= 0)
+    throw std::invalid_argument("numSplits must be greater than 0.");
+
   std::size_t const splitted_size = input_1d_vec.size() / numSplits;
-  // check splitted_size*numSplits == input_1d_vec.size()
+
+  if (splitted_size * numSplits != input_1d_vec.size())
+    throw std::invalid_argument("Conversion is not allowed! The input vector length " +
+                                std::to_string(input_1d_vec.size()) + " must be divisible by the numSplits " +
+                                std::to_string(numSplits) + ".");
+
   std::vector<std::vector<float>> output_2d_vec;
 
-  for (size_t i = 0; i < input_1d_vec.size(); i += numSplits - 1) {
-    std::vector<float> chunch_vec(input_1d_vec.begin() + i, input_1d_vec.begin() + i + splitted_size);
+  for (int i = 0; i < numSplits; i++) {
+    std::vector<float> chunch_vec(input_1d_vec.begin() + i * splitted_size,
+                                  input_1d_vec.begin() + (i + 1) * splitted_size);
     output_2d_vec.push_back(chunch_vec);
   }
   return output_2d_vec;
@@ -162,10 +171,11 @@ std::vector<float> OnlineDQMDigiAD::PrepareONNXDQMMapVectors(
 }
 
 std::vector<std::vector<std::vector<float>>> OnlineDQMDigiAD::ONNXOutputToDQMHistMap(
-    const std::vector<std::vector<float>> &ad_model_output_vectors, const int selOutputIdx) {
+    const std::vector<std::vector<float>> &ad_model_output_vectors,
+    const int numDepth,
+    const int numDIeta,
+    const int selOutputIdx) {
   // each output_vector is a serialized 3d hist map
-  const unsigned short numDepth = 7;
-  const unsigned short numDIeta = 64;
 
   const std::vector<float> &output_vector = ad_model_output_vectors[selOutputIdx];
   std::vector<std::vector<float>> output_2d_vec = Map1DTo2DVector(output_vector, numDepth);

--- a/DQM/HcalTasks/plugins/OnlineDQMDigiAD_cmssw.h
+++ b/DQM/HcalTasks/plugins/OnlineDQMDigiAD_cmssw.h
@@ -92,7 +92,10 @@ public:
     @return ad_model_output_vectors: the vectors of multidimensional arrays: output_data_0, output_data_1, ...
     */
   std::vector<std::vector<std::vector<float>>> ONNXOutputToDQMHistMap(
-      const std::vector<std::vector<float>> &ad_model_output_vectors, const int selOutputIdx = 7);
+      const std::vector<std::vector<float>> &ad_model_output_vectors,
+      const int numDepth,
+      const int numDIeta = 64,
+      const int selOutputIdx = 7);
 
 private:
   // onnx session

